aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -15,220 +15,443 @@ uid: fundamentals/servers/kestrel
+* `KestrelConfigurationLoader` mirrors the `Listen` family of APIs from `KestrelServerOptions` as `Endpoint` overloads, so code and config endpoints may be configured in the same place. These overloads don't use names and only consume default settings from configuration.
+
+*Change the defaults in code*
+
+`ConfigureEndpointDefaults` and `ConfigureHttpsDefaults` can be used to change default settings for `ListenOptions` and `HttpsConnectionAdapterOptions`, including overriding the default certificate specified in the prior scenario.
+
+```csharp
+options.ConfigureEndpointDefaults(opt =>
+{
+    opt.Protocols = HttpProtocols.Http1;
+});
+
+options.ConfigureHttpsDefaults(httpsOptions =>
+{
+    httpsOptions.SslProtocols = SslProtocols.Tls12;
+});
+```
+::: moniker-end
 
 **Bind to a TCP socket**
 
-The `Listen` method binds to a TCP socket, and an options lambda lets you configure an SSL certificate:
+The [Listen](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions.listen) method binds to a TCP socket, and an options lambda permits SSL certificate configuration:
 
-[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_DefaultBuilder&highlight=9-16)]
+[!code-csharp[](kestrel/samples/2.x/Program.cs?name=snippet_DefaultBuilder&highlight=9-16)]
 
-Notice how this example configures SSL for a particular endpoint by using [ListenOptions](https://github.com/aspnet/KestrelHttpServer/blob/rel/2.0.0/src/Microsoft.AspNetCore.Server.Kestrel.Core/ListenOptions.cs). You can use the same API to configure other Kestrel settings for particular endpoints.
+Notice how this example configures SSL for a particular endpoint by using [ListenOptions](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.listenoptions). Use the same API to configure other Kestrel settings for specific endpoints.
 
-[!INCLUDE [How to make an X.509 cert](../../includes/make-x509-cert.md)]
+[!INCLUDE [How to make an X.509 cert](~/includes/make-x509-cert.md)]
 
 **Bind to a Unix socket**
 
-You can listen on a Unix socket for improved performance with Nginx, as shown in this example:
+Listen on a Unix socket with [ListenUnixSocket](/dotnet/api/microsoft.aspnetcore.server.kestrel.core.kestrelserveroptions.listenunixsocket) for improved performance with Nginx, as shown in this example:
 
-[!code-csharp[](kestrel/sample2/Program.cs?name=snippet_UnixSocket)]
+[!code-csharp[](kestrel/samples/2.x/Program.cs?name=snippet_UnixSocket)]
 
 **Port 0**
 
-If you specify port number 0, Kestrel dynamically binds to an available port. The following example shows how to determine which port Kestrel actually bound to at runtime:
+When the port number is `0` is specified, Kestrel dynamically binds to an available port. The following example shows how to determine which port Kestrel actually bound at runtime:
+
+[!code-csharp[](kestrel/samples/2.x/Startup.cs?name=snippet_Configure&highlight=3-4,17-20)]
```

Comment:
This sample is pretty ironic because you wouldn't be able to connect to the site without knowing the port. The real port will be reported in the console output "Now listening on:". That might be better to show here.

Reply:
The port is never set to `0` anyway in the sample. I'll add a demo of it and the console window output.